packages/build/src/plugins_core/deploy_config/index.ts

```diff
@@ -0,0 +1,40 @@
+import { promises as fs } from 'fs'
+import { resolve } from 'path'
+
+import merge from 'deepmerge'
+
+import type { NetlifyConfig } from '../../index.js'
+import { CoreStep, CoreStepFunction } from '../types.js'
+
+const coreStep: CoreStepFunction = async function ({ buildDir, netlifyConfig }) {
+  const configPath = resolve(buildDir, '.netlify', 'deploy', 'v1', 'config.json')
+
+  let config: Partial<NetlifyConfig> = {}
+
+  try {
+    const data = await fs.readFile(configPath, 'utf8')
+
+    config = JSON.parse(data) as Partial<NetlifyConfig>
+  } catch {
+    return {}
+  }
+
+  const newConfig = merge(netlifyConfig, config)
```

Comment:
This is probably something you're already aware but if I recall correctly we have an allow list of fields that we allow to mutate:
- https://github.com/netlify/build/blob/ba81c85aad902380d24fc1722bc65d4feed24b84/packages/config/src/mutations/apply.js#L14-L80

Depending on the API we want to expose in the deploy config API we might need to revisit that list (or create some sort of exception for this particular step)

Reply:
Also, sort of related, if I'm understanding correctly here we're deep merging the config objects. After which ntl/config will take care of running through it's merge logic - https://github.com/netlify/build/blob/main/packages/config/src/merge.js - which takes care of removing things such as duplicate redirect entries and what not right? 🤔

Reply:
> This is probably something you're already aware but if I recall correctly we have an allow list of fields that we allow to mutate

Interestingly, I'm seeing that I'm able to mutate fields in the deploy configuration API that plugins shouldn't be able to mutate, which is a bit confusing to me since I'm using the exact same mechanism for mutating the config.

I'm working on that. ⏳

Reply:
> Also, sort of related, if I'm understanding correctly here we're deep merging the config objects. After which ntl/config will take care of running through it's merge logic - https://github.com/netlify/build/blob/main/packages/config/src/merge.js - which takes care of removing things such as duplicate redirect entries and what not right? 🤔

I think you're right, but I'm adding some tests to assert that.